pkg/performanceprofile/controller/performanceprofile_controller_hypershift.go

```diff
@@ -0,0 +1,650 @@
+		return err
+	}
+
+	ppConfigMap.Data[tunedConfigMapConfigKey] = string(ppEncoded)
+
+	//set label with performance profile name to ease the delete process
+	if ppConfigMap.Labels == nil {
+		ppConfigMap.Labels = make(map[string]string)
+	}
+	ppConfigMap.Labels[hypershiftPerformanceProfileNameLabel] = performanceProfile.Name
+
+	if err := cli.Update(ctx, ppConfigMap); err != nil {
+		klog.Errorf("failed to update performance profile from configMap %q: %v", ppConfigMap.Name, err)
+		return err
+	}
+
+	return nil
+}
+
+func hypershiftDeleteComponents(remoteClient client.Client, ctx context.Context, ppConfigMap *corev1.ConfigMap) error {
+	// ConfigMap is marked for deletion and waiting for finalizers to be empty
+	// so better to clean-up and delete the objects.
+
+	if ppConfigMap.Labels == nil {
+		return fmt.Errorf("ConfigMap %s has no Labels.Unable to finalize deletion procedure properly", ppConfigMap.Name)
+	}
+
+	performanceProfileName, ok := ppConfigMap.Labels[hypershiftPerformanceProfileNameLabel]
+	if !ok {
+		return fmt.Errorf("ConfigMap %s has no label %q. Unable to finalize deletion procedure properly", ppConfigMap.Name, hypershiftPerformanceProfileNameLabel)
+	}
+
+	// just delete RtClass, as right now all the other components created from this PP
+	// are embedded into ConfigMaps which has an OwnerReference with the PP configmap
+	// and will be deleted by k8s machinery trigerring the deletion procedure of the
+	// embedded elements.
+	rtName := runtimeclass.BuildRuntimeClassName(performanceProfileName)
+	rtClass, err := readRuntimeClass(remoteClient, ctx, rtName)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			// rtClass not found so nothing to delete, so delete process has finished
+			return nil
+		}
+		return fmt.Errorf("unable to read RuntimeClass %q, error: %w. Unable to finalize deletion procedure properly", rtName, err)
+	}
+
+	err = remoteClient.Delete(ctx, rtClass)
+	return err
+}
+
+func configMapHasFinalizer(cm *corev1.ConfigMap, finalizer string) bool {
+	for _, f := range cm.Finalizers {
+		if f == finalizer {
+			return true
+		}
+	}
+	return false
+}
+
+func configMapRemoveFinalizer(cm *corev1.ConfigMap, finalizer string) *corev1.ConfigMap {
+	var finalizers []string
+	for _, finalizer := range cm.Finalizers {
+		if finalizer != hypershiftFinalizer {
+			finalizers = append(finalizers, finalizer)
+		}
+	}
+	cm.Finalizers = finalizers
+
+	return cm
+}
+
+func hashStruct(o interface{}) string {
+	hash := fnv.New32a()
+	hash.Write([]byte(fmt.Sprintf("%v", o)))
+	intHash := hash.Sum32()
+	return fmt.Sprintf("%08x", intHash)
+}
+
+// parseNamespacedName expects a string with the format "namespace/name"
+// and returns the name only.
+// If given a string in the format "name" returns "name".
+func parseNamespacedName(namespacedName string) string {
+	parts := strings.SplitN(namespacedName, "/", 2)
+	if len(parts) > 1 {
+		return parts[1]
+	}
+	return parts[0]
+}
```

Comment:
Could be moved to a shared hypershift pkg

Reply:
Handle this as part of #529